x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/utils/merge_esql_result_in_source.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SignalSource } from '../../types';
+import {
+  robustGet,
+  robustUnset,
+} from '../../utils/source_fields_merging/utils/robust_field_access';
+
+export const mergeEsqlResultInSource = (
```

Comment:
This might merit some documentation, especially to describe the purpose (e.g. "used to ensure that any non-conflicting fields in the ES|QL response are added to the generated alert")